config_ipa.go

```diff
@@ -46,41 +46,34 @@ const (
 	EmptyCodeHashSecondHalfIdx = EmptyCodeHashFirstHalfIdx + 1
 )
 
-type IPAConfig struct {
-	conf *ipa.IPAConfig
-}
+var (
+	FrZero Fr
+	FrOne  Fr
 
-type Config = IPAConfig
+	cfg     *Config
+	onceCfg sync.Once
+)
 
-func (ipac *IPAConfig) CommitToPoly(poly []Fr, _ int) *Point {
-	ret := ipac.conf.Commit(poly)
-	return &ret
+func init() {
+	FrZero.SetZero()
+	FrOne.SetOne()
 }
 
-var cfg *Config
+type IPAConfig struct {
+	conf *ipa.IPAConfig
+}
 
-var precompFileName = "precomp"
+type Config = IPAConfig
 
 func GetConfig() *Config {
-	if cfg == nil {
-		var ipacfg *ipa.IPAConfig
-		if precompSer, err := os.ReadFile(precompFileName); err != nil {
-			ipacfg = ipa.NewIPASettings()
-			serialized, err := ipacfg.SRSPrecompPoints.SerializeSRSPrecomp()
-			if err != nil {
-				panic("error writing serialized precomputed Lagrange points:" + err.Error())
-			} else if err = os.WriteFile(precompFileName, serialized, 0666); err != nil {
-				panic("error saving the precomp: " + err.Error())
-			}
-		} else {
-			srs, err := ipa.DeserializeSRSPrecomp(precompSer)
-			if err != nil {
-				panic("error deserializing precomputed Lagrange points:" + err.Error())
-			}
-			ipacfg = ipa.NewIPASettingsWithSRSPrecomp(srs)
+	onceCfg.Do(func() {
```

Comment:
I prefer to be safe and use `sync.Once` here. Our previous way can be racy and trust the client to not fire two goroutines that would call `GetConfig()` and mess everything up.

It also allows the tests to run in parallel since it doesn't care who calls (and if done in parallel) `GetConfig()`.